File: NetFramework/FlexGrid/CS/RowDetails/Samples/FlexGridSample.cs
using System;
using System.Data;
using System.Windows.Forms;
using C1.Win.C1FlexGrid;

namespace RowDetails
{
    public partial class FlexGridSample : UserControl
    {
        public FlexGridSample()
        {
            InitializeComponent();
        }

        private void FlexGrid_Load(object sender, EventArgs e)
        {
            if (DesignMode) return;

            string conn = Util.GetConnectionString();
            var ds = new DataSet();
            string[] tables = "Customers, Orders".Split(',');
            foreach (string tableName in tables)
            {
                Util.FillTable(ds, tableName, conn);
            }

            ds.Relations.Add("Customers_Orders",
                ds.Tables["Customers"].Columns["CustomerID"],
                ds.Tables["Orders"].Columns["CustomerID"]);

            flexGrid.DataSource = ds;
            flexGrid.DataMember = "Customers";
            flexGrid.RowDetailProvider = (g, r) => new C1FlexGridRowDetail();
            flexGrid.AreRowDetailsFrozen = false;
        }
    }
}

File: NetFramework/FlexGrid/VB/RowDetails/Samples/FlexGridSample.vb
Imports C1.Win.FlexGrid

Public Class FlexGridSample
    Private Sub FlexGridSample_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim conn As String = Util.GetConnectionString()
        Dim ds = New DataSet()
        Dim tables As String() = "Customers, Orders".Split(","c)
        For Each tableName As String In tables
            Util.FillTable(ds, tableName, conn)
        Next

        ds.Relations.Add("Customers_Orders", ds.Tables("Customers").Columns("CustomerID"), ds.Tables("Orders").Columns("CustomerID"))

        flexGrid.DataSource = ds
        flexGrid.DataMember = "Customers"

        flexGrid.RowDetailProvider = Function(g, r) New C1FlexGridRowDetail()
        flexGrid.AreRowDetailsFrozen = False
    End Sub
End Class
